pytest/lib/bridge.py

```diff
@@ -1,14 +1,142 @@
+from enum import Enum
+import json
+import logging
+from multiprocessing import Process, Value
 from pathlib import Path
+import random
+from retrying import retry
 import signal
 import shutil
+import string
 import subprocess
 import time
 import traceback
 import os
 
+from key import Key
+from messages.bridge import *
+from serializer import BinarySerializer
+from transaction import sign_function_call_tx, sign_create_account_with_full_access_key_and_balance_tx
+
+logger = logging.getLogger('bridge')
+log_handler = logging.StreamHandler()
+log_handler.setFormatter(logging.Formatter('[%(asctime)s] %(levelname)-8s %(process)-6d %(funcName)-18s %(message)s', '%Y-%m-%d %H:%M:%S'))
+# do not propagate bridge messages to the root logger
+logger.propagate = False
+logger.addHandler(log_handler)
+logging.getLogger('bridge').setLevel(logging.INFO)
+
+MAX_ATTEMPTS = 10
+
+bridge_cluster_config_changes = {
+    "consensus": {
+        "min_block_production_delay": {
+            "secs": 4,
+            "nanos": 0,
+        },
+        "max_block_production_delay": {
+            "secs": 8,
+            "nanos": 0,
+        },
+        "max_block_wait_delay": {
+            "secs": 24,
+            "nanos": 0,
+        },
+    }
+}
+
+class JSAdapter:
+
+    def __init__(self, config):
+        self.config = config
+        self.bridge_dir = self.config['bridge_dir']
+        self.cli_dir = os.path.join(self.bridge_dir, 'cli')
+
+    def js_call(self, args, timeout):
+        logger.info(' '.join(args))
+        try:
+            res = subprocess.check_output(args, timeout=timeout, stderr=subprocess.STDOUT, cwd=self.cli_dir).decode('ascii').strip()
+            return res
+        except subprocess.CalledProcessError as e:
+            logger.error('js call failed, exit code %d, msg: %s' % (e.returncode, e.output))
+            assert False
+
+    def js_call_cli(self, args, timeout=3):
+        if not isinstance(args, list):
+            args = [args]
+        args = ['node', 'index.js'] + args
+        return self.js_call(args, timeout)
+
+    def js_call_testing(self, args, timeout=3):
+        if not isinstance(args, list):
+            args = [args]
+        args = ['node', 'index.js', 'TESTING'] + args
+        return self.js_call(args, timeout)
+
+    def js_call_service(self, args, stdout, stderr):
+        if not isinstance(args, list):
+            args = [args]
+        args = ['node', 'index.js', 'start'] + args + ['--daemon', 'false']
+        logger.info(' '.join(args))
+        service = subprocess.Popen(args, stdout=stdout, stderr=stderr, cwd=self.cli_dir)
+        # TODO ping and wait until service really starts
+        time.sleep(5)
+        return service
+
+
+class BridgeUser(object):
+
+    def __init__(self, name, eth_secret_key, near_sk, near_pk):
+        self.name = name
+        self.eth_secret_key = eth_secret_key
+        self.near_account_name = name + '.test0'
+        self.near_sk = near_sk
+        self.near_pk = near_pk
```

Comment:
We can reuse the class `Key` from: https://github.com/near/nearcore/blob/49048c733c08e2561db3c8b106c0248f32f479df/pytest/lib/key.py#L5

Reply:
++